irohad/ametsuchi/impl/rocksdb_common.hpp

```diff
@@ -683,6 +683,8 @@ namespace iroha::ametsuchi {
     template <typename F, typename S, typename... Args>
     auto enumerate(F &&func, S const &fmtstring, Args &&... args) {
       auto it = seek(fmtstring, std::forward<Args>(args)...);
+      static_assert(std::is_convertible_v<std::result_of_t<F&(decltype(it),size_t)>, bool>,
+                    "Must F(unique_ptr<rocksdb::Iterator>,size_t) -> bool");
```

Comment:
What the point of current assert? If F will not be convertible it doesn't compile at all.

Reply:
To reduce compile-time listing

Reply:
discussed